File: codeforces/1305B.py
'''
	CodeForces 1305B, Ozon Tech Challenge 2020
	Kuroni and Simple Strings

	Tags: String Manipulation, Greedy
'''
from itertools import accumulate

def solve(p, L, R):
	if p[1] is '(' and L <= R:
		return True
	if p[1] is ')' and L >= R:
		return True
	return False

s = input()

left  = list(accumulate([ 1 if ch is '(' else 0 for ch in s ]))
right = list(reversed(list(accumulate([ 1 if ch is ')' else 0 for ch in reversed(s) ]))))

ans = list(map(lambda x: str(x[0] + 1), filter(lambda p: solve(p, left[p[0]], right[p[0]]), enumerate(s))))

if len(ans) == 0:
	print('0')
else:
	print('1')
	print(len(ans))
	print(' '.join(ans))

File: since2020/CodeForces/1305B.cpp
/**
 *  @judge CodeForces
 *  @id 1305B
 *  @name Kuroni and Simple Strings
 *  @contest Ozon Tech Challenge 2020
 * 
 *  @tag String Manipulation, Greedy
 */
#include <iostream>
using namespace std;

#define MAX 1010

string s;
int LL[MAX], RR[MAX];

int check(char A, char B) {
	if (A == B)
		return 1;
	return 0;
}

void solve() {
	LL[0] = check(s[0], '(');
	RR[s.size() - 1] = check(s[s.size() - 1], ')');
	for (int i = 1; i < s.size(); i++) {
		int j = s.size() - 1 - i;
		LL[i] = LL[i - 1] + check(s[i], '(');
		RR[j] = RR[j + 1] + check(s[j], ')');
	}
	for (int i = 0; i < s.size() - 1; i++) {
		if (LL[i] > 0 && LL[i] == RR[i + 1]) {
			cout << '1' << endl;
			cout << LL[i] * 2 << endl;
			int count = 0;
			for (int j = 0; j <= i; j++) {
				if (s[j] == '(') {
					if (count)
						cout << " ";
					cout << j + 1;
					count++;
				}
			}
			for (int j = i + 1; j < s.size(); j++) {
				if (s[j] == ')')
					cout << " " << j + 1;
			}
			cout << endl;
			return ;
		}
	}
	cout << '0' << endl;
	return ;
}

int main() {
	cin >> s;
	solve();
}